### PR TITLE
Migrate rocm test to using oidc

### PR DESCRIPTION
**Pull request by atalman**
Similar to Intel XPU, lets use OIDC for rocm runners.

Refer to this PR: https://github.com/pytorch/pytorch/pull/116554

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
LGTM! Thank you for the quick follow-up.  Having a role ready helps :)

File: .github/workflows/_rocm-test.yml [hunk 1]
```diff
@@ -61,6 +65,17 @@ jobs:
       - name: Setup ROCm
         uses: ./.github/actions/setup-rocm
 
+      - name: configure aws credentials
+        id: aws_creds
+        uses: aws-actions/configure-aws-credentials@v1.7.0
```
Comment (malfet) [inline .github/workflows/_rocm-test.yml:70]: 1.7.0 were release 2+ years ago, is there a reason not to migrate to the latest release?
```suggestion
        uses: aws-actions/configure-aws-credentials@v4
```

File: .github/workflows/_rocm-test.yml [hunk 2]
```diff
@@ -61,6 +65,17 @@ jobs:
...
+        with:
+          role-to-assume: arn:aws:iam::308535385114:role/gha_workflow_pytorch_artifacts
```
Comment (malfet) [inline .github/workflows/_rocm-test.yml:72]: Please mention in PR description, what privileges this role gives to the runner?  Right now I see it gives one ability to ecr:PutImage, which is not necessary to run tests, is it?

**Review by malfet — CHANGES_REQUESTED** [review]
Assigned role gives one ability to upload new images to elastic container registry, but testers should only be able to read those, don't they?

File: .github/workflows/_rocm-test.yml (hunk 2 above)
Comment (atalman) [reply .github/workflows/_rocm-test.yml:72]: Thank you. Adjusted this role to allow read only ecr actions.

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**